mypy/subtypes.py

```diff
@@ -392,6 +392,7 @@ def f(self) -> A: ...
     as well.
     """
     assert right.type.is_protocol
+    TypeState.record_protocol_subtype_check(left.type, right.type)
```

Comment:
Please add short comment explaining why we do this ("We need to record this check to generate fine-grained dependencies." or something).